File: LudoGame/LudoGame/Models/LudoGameState.swift
import Foundation

struct LudoGameState: Equatable, Codable {
    var players: [GamePlayer]
    var currentPlayerIndex: Int
    var lastDiceRoll: Int?
    var consecutiveSixes: Int
    var selectableTokenIDs: Set<Int>
    var statusMessage: String
    var winner: PlayerColor?
    var isRolling: Bool

    init(playerCount: Int) {
        let colors = Array(PlayerColor.allCases.prefix(playerCount))
        self.players = colors.map { GamePlayer(color: $0) }
        self.currentPlayerIndex = 0
        self.lastDiceRoll = nil
        self.consecutiveSixes = 0
        self.selectableTokenIDs = []
        self.statusMessage = "Roll the dice to begin"
        self.winner = nil
        self.isRolling = false
    }

    var currentPlayer: GamePlayer {
        players[currentPlayerIndex]
    }

    var activePlayers: [GamePlayer] {
        players.filter(\.isActive)
    }
}

File: LudoGameAndroid/app/src/main/java/com/ludogame/app/models/LudoGameState.kt
package com.ludogame.app.models

data class LudoGameState(
    val players: List<GamePlayer>,
    val currentPlayerIndex: Int = 0,
    val lastDiceRoll: Int? = null,
    val consecutiveSixes: Int = 0,
    val selectableTokenIDs: Set<Int> = emptySet(),
    val statusMessage: String = "Roll the dice to begin",
    val winner: PlayerColor? = null,
    val isRolling: Boolean = false
) {
    constructor(playerCount: Int) : this(
        players = PlayerColor.allCases.take(playerCount.coerceIn(2, 4)).map { GamePlayer(it) }
    )

    val currentPlayer: GamePlayer
        get() = players[currentPlayerIndex]
}
